src/Microsoft.IdentityModel.Protocols/Configuration/HttpDocumentRetriever.cs

```diff
@@ -87,6 +97,9 @@ public async Task<string> GetDocumentAsync(string address, CancellationToken can
             {
                 LogHelper.LogVerbose(LogMessages.IDX20805, address);
                 var httpClient = _httpClient ?? _defaultHttpClient;
+                if (AllowAdditionalHeaderData && SendAdditionalHeaderData)
+                    IdentityModelTelemetryUtils.SetTelemetryData(httpClient);
```

Comment:
Do not manipulate DefaultRequestHeaders, you don't own it. It's also incorrect to modify it per request.

Instead, change the below code to create a new HttpRequestMessage and add the headers there.

Reply:
@Tratcher
 Makes sense. This was only a draft PR to demonstrate an idea. I will go ahead and abandon this one, Original PR is [here](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/1632)